src/testing/test-helper.js

```diff
@@ -285,7 +285,7 @@ function verifyChannelAssignment(expectedChannels) {
 }
 
 function checkAuthorizations(expectedAuthorizations, actualAuthorizations, authorizationType) {
-  if (!(expectedAuthorizations instanceof Array)) {
+  if (!Array.isArray(expectedAuthorizations)) {
```

Comment:
As mentioned in the pull request description, `Array.isArray` is the most reliable way to determine whether something is an array. Even though this function was added in ECMAScript 5 and Sync Gateway's JavaScript engine does not fully support all of ES5, I have manually verified that it does at least support `Array.isArray`.